teps/0089-nonfalsifiable-provenance-support.md

```diff
@@ -127,28 +134,91 @@ containers:
+Each time the TaskRun needs to be modified,
+* Controller verifies the TaskRun status hasn't been modified
+* Controller requests a SPIRE SVID and then uses it to sign the new modified TaskRun
 * Controller stores the new SVID x509 and signature on the TaskRun as an annotation
 
-Right now, the design details around this are a little fuzzy!
-The plan is to first implement signed Results, and then if this design no longer seems like the correct fit for signed TaskRuns then it will be revisited in this TEP :)
-
 Things to keep in mind when designing this feature further:
 * There could potentially be a performance implication here, if we need to request signatures every time the TaskRun is modified during execution.
-* We may want to ignore labels/annotations when checking for modifications, since they might be updated by other services
+* What are the fields which are important to detect tampering, ensuring a level of flexibility for operational  usecases - i.e. other operators.
 * What if mutating admission controllers are intentionally changing TaskRuns (e.g. Solarwinds injecting Tasks into a Pipeline)
+* Can we assert that we are verifying the signatures against the proper authority, what are potential threat vectors here.
 
 ### 2. Tekton Pipelines can't verify that the results it reads weren't modified
 The solution to this is Signed Results.
 We will modifiy the entrypointer image to sign results with SPIRE once they're available.
 The signature and SVID provided by SPIRE will be stored on the TaskRun itself as annotations for a client to verify (this could be Pipelines or Chains).
 
+### Architecture
+
+Here is a brief overview of the architecture for Tekton pipelines and SPIRE.
+```
+┌─────────────┐  Register TaskRun Workload Identity           ┌──────────┐
+│             ├──────────────────────────────────────────────►│          │
+│  Tekton     │                                               │  SPIRE   │
+│  Controller │◄───────────┐                                  │  Server  │
+│             │            │ Listen on TaskRun                │          │
+└────────────┬┘◄┐          │                                  └──────────┘
+ ▲           │  │  ┌───────┴───────────────────────────────┐     ▲
+ │           │  │  │           Tekton TaskRun              │     │
+ │           │  │  └───────────────────────────────────────┘     │
+ │  Configure│  │                                                │ Attest
+ │  Pod &    │  └─────────────────┐ TaskRun Entrypointer         │   +
+ │  check    │                    │ Sign Result and update       │ Request
+ │  ready    │     ┌───────────┐  │ the status with the          │ SVIDs
+ │           └────►│  TaskRun  ├──┘ signature + cert             │
+ │                 │  Pod      │    which will be used by        │
+ │                 └───────────┘    TektonController to          │
+ │                   ▲              update TaskRun.              │
```

Comment:
two things im wondering, for the 2nd im probably just not understanding a detail of how this all works:

1. how does the entrypointer provide the signature and cert? (how large are they, are they small enough to be transmitted via the termination message, and how much space will that leave for the results themselves)
2. why does the entrypointer provide the cert - i would have thought the tekton controller would want to request the cert from SPIRE independently, so it knows it can trust it?

Reply:
> The size of the cert is about 800 bytes, and the size of the signatures is about 100 bytes * (number of fields + 1).

How important is it that the cert itself is emitted from the TaskRun? Is it possible we can represent the chain of trust some other way? (e.g. some unique identifier for the pod?)

While we are trying to add support for large results ([TEP-0086](https://github.com/tektoncd/community/pull/521)), at the moment we're limited to a termination message of 4k  (https://github.com/tektoncd/community/blob/main/teps/0076-array-result-types.md#size-limits) - so we're looking at taking up 20% of that space with the cert. (not to mention that this is nearly 1k worth of extra data in every single taskrun?) I suppose that could be worse but I'm wondering if we absolutely _need_ to include it

(and assuming we do need it, is it possible to add some details to the requirements section around this?)

Reply:
> upload it to through the entrypointer in the pod

Is it possible to utilize "reserved directories" for this purpose? https://github.com/tektoncd/pipeline/tree/main/docs/developers#tekton

Reply:
I agree with @lumjjb. I did not see anything else in the code related to that. The termination message from each container within the pod was the method it used to get the TaskRun results back out. These are parsed and filtered to show as TaskRun Results.